### ISSUE TITLE
disallow authenticating post-registration?

### ISSUE THREAD

**Issue by slingamn** [ergochat/ergo#1109]
This is a discussion issue; I expect this to be controversial. I'm proposing that we remove any way to log into an account post-registration, except for `NS VERIFY`. In particular, `NS IDENTIFY` would be removed, and `AUTHENTICATE` would be disallowed for registered clients. Clients would have to authenticate either with SASL, or with the new PASS-stuffing technique from #1020.

The motivation for this is to reduce edge cases. For example, logging into an account when `force-nick-equals-account` is enabled, but an existing client is holding the relevant nickname: we currently respond by rolling back the login. #1108 is similar.

**Comment by slingamn** [comment]
Another interesting consequence of this would be that we would never need to send account-notify --- we could keep advertising the cap and ACK'ing requests for it, but we would never need to send a message.

**Comment by DanielOaks** [comment]
I'm not sure how I feel about this. I'd be interested to hear from some client devs and get their thoughts on it. The main issue with this for me is that the best resource we can link users to when they try `NS IDENTIFY` and it fails is https://freenode.net/kb/answer/sasl and I'd rather not link to that page from the software. I don't thiiiink I mind only allowing SASL during connection reg, but it does feel pretty alien.

**Comment by slingamn** [comment]
We could link to our own manual in this case maybe?

**Comment by KoraggKnightWolf** [comment]
Perhaps the best way would be a configuration option? like disallow-post-authentication true|false might be a solid middleground.
Personally I would be against such a rather drastic measure as even though it SHOULD be trivial enough to setup SASL PLAIN (and with it SSL to be safe), real world experiences show that people often struggle even just getting SSL to work, let alone SASL. 
Also many like to use outdated (mIRC) scripts which would lack support for it (another thing is that many are told PASS is for server passwords so they might not understand this can also be used for sercies authing)

**Comment by DanielOaks** [comment]
Yeah the point of doing this would mostly be to remove the code cruft from letting users login after connecting (lotsa state to change and messages to send), but even most of that won't be able to be removed because registering/verifying your account will result in you being logged-in and I don't think we can reasonably remove that.

@slingamn the point about it is more that we need an itemised list of clients and client-specific instructions on how to setup/use SASL, since... there's no real standard "just click XYZ button and do it" and telling users to google "whateverclient SASL" is really user-unfriendly. if we were building a resource like this it'd need to be some separate pages on our site or whatever that we can link to in future (I've been planning to build something similar for the [ircdocs compendium site](https://github.com/ircdocs/compendium/) a while back but haven't actually made the page yet.
